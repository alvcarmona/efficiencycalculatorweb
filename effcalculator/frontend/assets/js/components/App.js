import React, { Component } from 'react';
import Homepage from  './homepage'
import DetectorEfficiencyCalculator from '../components/DetectorEfficiencyCalculator';
import logo from '../../img/logo.svg';
import '../../styles/App.css';
import { Route, Switch, Link } from 'react-router-dom'
import { Navbar, NavItem, NavDropdown, Nav, MenuItem } from 'react-bootstrap'

class App extends Component {

  render() {
    return (

      <div className="App">

        <div>
             <Navbar inverse collapseOnSelect>
                    <Navbar.Header>
                        <Navbar.Brand>
                            <Link to={`/frontend/homepage/`}>Eff calculator</Link>
                        </Navbar.Brand>
                        <Navbar.Toggle />
                    </Navbar.Header>
                    <Navbar.Collapse>
                        <Nav>
                            <NavItem eventKey={1} href="/detectors"><Link to={`/frontend/detectors/`}>Detectors</Link></NavItem>
                      </Nav>
                    </Navbar.Collapse>
                </Navbar>
             <Switch>
                 <Route path='/frontend/homepage' component={Homepage}/>
                <Route path='/frontend/detectors' component={DetectorEfficiencyCalculator}/>
             </Switch>
        </div>
      </div>
    );
  }
}



export default App;