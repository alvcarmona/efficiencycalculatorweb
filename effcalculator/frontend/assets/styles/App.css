@import url("//netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap-glyphicons.css");

li {
    list-style: none;
}

.App {
    text-align: center;
    width: 100%;
    height: 100%;
}

.App-logo {
    animation: App-logo-spin infinite 20s linear;
    height: 80px;
}

.App-header {
    background-color: #222;
    height: 150px;
    padding: 20px;
    color: white;
}

.App-intro {
    font-size: large;
}

@keyframes App-logo-spin {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}

p.DetectorLinks a {
    margin: 5px;
}

.masonryElement {
    margin: 10px;
}

.jumbotron:hover {
    background-color: #b1daff !important;
    -webkit-box-shadow: 0px 2px 123px -17px rgba(0, 0, 0, 0.75);
    -moz-box-shadow: 0px 2px 123px -17px rgba(0, 0, 0, 0.75);
    box-shadow: 0px 2px 123px -17px rgba(0, 0, 0, 0.75);
}

.jumbotron {
    background-color: #fff !important;
    -webkit-box-shadow: 0px 2px 21px -2px rgba(0, 0, 0, 0.75);
    -moz-box-shadow: 0px 2px 21px -2px rgba(0, 0, 0, 0.75);
    box-shadow: 0px 2px 21px -2px rgba(0, 0, 0, 0.75);
    -webkit-transition: background-color 0.5s, -moz-box-shadow 0.5s, box-shadow 0.5s, -webkit-box-shadow 0.5s; /* Safari */
    transition: background-color 1s, -moz-box-shadow 1s, box-shadow 1s, -webkit-box-shadow 1s;

}

.DetectorDetail .Container {
    margin-top: 10px;
    padding-top: 10px;
    padding-bottom: 10px;
    background-color: white;
    -webkit-box-shadow: 0px 2px 21px -2px rgba(0, 0, 0, 0.75);
    -moz-box-shadow: 0px 2px 21px -2px rgba(0, 0, 0, 0.75);
    box-shadow: 0px 2px 21px -2px rgba(0, 0, 0, 0.75);
}

.DetectorEditComponent .Container {
    margin-top: 15px;
    padding-top: 15px;
    padding-bottom: 15px;
    background-color: white;
    -webkit-box-shadow: 0px 2px 21px -2px rgba(0, 0, 0, 0.75);
    -moz-box-shadow: 0px 2px 21px -2px rgba(0, 0, 0, 0.75);
    box-shadow: 0px 2px 21px -2px rgba(0, 0, 0, 0.75);
}

.DetectorEfficiencyComponent .Container {
    margin-top: 15px;
    padding: 50px;
    background-color: white;
    -webkit-box-shadow: 0px 2px 21px -2px rgba(0, 0, 0, 0.75);
    -moz-box-shadow: 0px 2px 21px -2px rgba(0, 0, 0, 0.75);
    box-shadow: 0px 2px 21px -2px rgba(0, 0, 0, 0.75);
}

.body {
    background-color: #f3f7f2;
}

.brand-primary {
    background-color: #60c3ed !important;
}



ul.my-Gallery-class {
    margin-top: 35px;
}

.createButton {
    margin-top: 20px;
}

div.plotRow.row {
    margin: 5px;
}

.submitButton {
    margin: 15px;
}

.danger-config {
    margin: 100px;
}

/************************* Ribbon *****************************************/

body:after {
    content: "beta";
    position: fixed;
    width: 80px;
    height: 25px;
    background: #EE8E4A;
    top: 7px;
    left: -20px;
    text-align: center;
    font-size: 13px;
    font-family: sans-serif;
    text-transform: uppercase;
    font-weight: bold;
    color: #fff;
    line-height: 27px;
    -ms-transform: rotate(-45deg);
    -webkit-transform: rotate(-45deg);
    transform: rotate(-45deg);
}

/************************* homepage *****************************************/

.center-parent {
  position: relative;
}
.center-child {
  align-self: center;
}

.brightness {
    background-color: #9bbe05 !important;
    padding: 5px;
}

.brightnessImg {
    display: inline-block !important;;
}

.homepage-container {
    width: 100% !important;
    padding: 0px !important;
}

#header-button {
    margin: 10px;
}

.bg-image {
    background: #1c262b;
    text-align: center;
    position: relative;
    z-index: 1;
    display: block;
    overflow: hidden;
    padding: 10px;
}

.githubSection{
        margin-top: -20px;
}
/*
div.bg-image.row {
    height: 45%;
}
*/
.header-content {
    margin-top: 10%;

}

h2.homepageheader {
    color: #f9f9f9;
}

.bg-image:before {
    content: '';
    display: block;
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    width: 100%;
    height: 100%;
    z-index: -1;
    background: #333 url("https://project.esss.dk/owncloud/index.php/apps/files_sharing/ajax/publicpreview.php?x=1732&y=610&a=true&file=ess_big_view_External%2520Use.jpg&t=RlogvXmGTiKkKD4&scalingup=0") no-repeat top center;
    background-size: 100% 50%;
    background-attachment: fixed;
    -webkit-filter: blur(5px);
    filter: blur(5px);
    opacity: .8;
    -webkit-transform: scale(1.1);
    transform: scale(1.1);
}

.bg-image.bg-image-2:before {
    opacity: .6;
    background-image: url("https://s3-us-west-2.amazonaws.com/s.cdpn.io/366117/8_copy_copy.jpg");
    background-position: center center;
}

.description {
    width: 100%;
    height: 20%;
    text-align: justify;
    position: relative;
}

.spacerHome-alt {
     padding: 35px 50px 0px 10%;
    background-color: #f9f9f9;
}

.spacerHome {
     padding: 35px 50px 0px 10%;
}
/************************* Explanation *****************************************/
.Explanation{
    margin-top: 50px;
    margin-bottom: 50px;
}


/************************* Edit *****************************************/

.edit-form {
    margin: 25px;
}
.error-message{
    color:red;
}
.form-field{
    margin-top: 15px;
}

.center-parent {
  position: relative;
}
.modal-body {
  padding-left: 50px !important;
}




/************************* Navbar *****************************************/
nav.brand-primary {
    -webkit-box-shadow: 0px 2px 21px -2px rgba(0, 0, 0, 0.75);
    -moz-box-shadow: 0px 2px 21px -2px rgba(0, 0, 0, 0.75);
    box-shadow: 0px 2px 21px -2px rgba(0, 0, 0, 0.75);
    border: 0px !important;
}

.navbar {
    margin-bottom: 0px !important;
}

a.navbar-brand {
    color: white !important;
}

a.navbar-brand:hover {
    color: lightgrey !important;
}

.NavbarItem:hover {
    background-color: #5fadd7 !important;
}

.detectordetail.row {
    margin-top: 30px;
}

a.NavbarLink {
    color: white !important;
}

a.NavbarLink:hover {
    color: #c9c9c9 !important;
}

/************************* Detail *****************************************/

.BladeThicknessPlot {
    margin: 25px;
}

.BladeThicknessTable {
    margin: 20px;
    margin-top: 55px;
}

.parametercol{
    margin-top: 65px;
    margin-left: 45px;
}

.refreshDetectorsButton{
     margin-top: 1px;
     margin-right: 2px;
     position:absolute;
     top:80px;
     right:20px;
 }
#download{
    font-size: 16px
}

/*********************Sketch*********************************/
#plano {
    width: 50%;
  margin: 0 auto;
  display: block;
  border: 1px solid #d9d9d9;
}
.sketch{
    margin:50px;
}



/************************* Loading spinner *****************************************/

.sk-fading-circle {
    margin: 100px auto;
    width: 40px;
    height: 40px;
    position: relative;
}

.sk-fading-circle .sk-circle {
    width: 100%;
    height: 100%;
    position: absolute;
    left: 0;
    top: 0;
}

.sk-fading-circle .sk-circle:before {
    content: '';
    display: block;
    margin: 0 auto;
    width: 15%;
    height: 15%;
    background-color: #333;
    border-radius: 100%;
    -webkit-animation: sk-circleFadeDelay 1.2s infinite ease-in-out both;
    animation: sk-circleFadeDelay 1.2s infinite ease-in-out both;
}

.sk-fading-circle .sk-circle2 {
    -webkit-transform: rotate(30deg);
    -ms-transform: rotate(30deg);
    transform: rotate(30deg);
}

.sk-fading-circle .sk-circle3 {
    -webkit-transform: rotate(60deg);
    -ms-transform: rotate(60deg);
    transform: rotate(60deg);
}

.sk-fading-circle .sk-circle4 {
    -webkit-transform: rotate(90deg);
    -ms-transform: rotate(90deg);
    transform: rotate(90deg);
}

.sk-fading-circle .sk-circle5 {
    -webkit-transform: rotate(120deg);
    -ms-transform: rotate(120deg);
    transform: rotate(120deg);
}

.sk-fading-circle .sk-circle6 {
    -webkit-transform: rotate(150deg);
    -ms-transform: rotate(150deg);
    transform: rotate(150deg);
}

.sk-fading-circle .sk-circle7 {
    -webkit-transform: rotate(180deg);
    -ms-transform: rotate(180deg);
    transform: rotate(180deg);
}

.sk-fading-circle .sk-circle8 {
    -webkit-transform: rotate(210deg);
    -ms-transform: rotate(210deg);
    transform: rotate(210deg);
}

.sk-fading-circle .sk-circle9 {
    -webkit-transform: rotate(240deg);
    -ms-transform: rotate(240deg);
    transform: rotate(240deg);
}

.sk-fading-circle .sk-circle10 {
    -webkit-transform: rotate(270deg);
    -ms-transform: rotate(270deg);
    transform: rotate(270deg);
}

.sk-fading-circle .sk-circle11 {
    -webkit-transform: rotate(300deg);
    -ms-transform: rotate(300deg);
    transform: rotate(300deg);
}

.sk-fading-circle .sk-circle12 {
    -webkit-transform: rotate(330deg);
    -ms-transform: rotate(330deg);
    transform: rotate(330deg);
}

.sk-fading-circle .sk-circle2:before {
    -webkit-animation-delay: -1.1s;
    animation-delay: -1.1s;
}

.sk-fading-circle .sk-circle3:before {
    -webkit-animation-delay: -1s;
    animation-delay: -1s;
}

.sk-fading-circle .sk-circle4:before {
    -webkit-animation-delay: -0.9s;
    animation-delay: -0.9s;
}

.sk-fading-circle .sk-circle5:before {
    -webkit-animation-delay: -0.8s;
    animation-delay: -0.8s;
}

.sk-fading-circle .sk-circle6:before {
    -webkit-animation-delay: -0.7s;
    animation-delay: -0.7s;
}

.sk-fading-circle .sk-circle7:before {
    -webkit-animation-delay: -0.6s;
    animation-delay: -0.6s;
}

.sk-fading-circle .sk-circle8:before {
    -webkit-animation-delay: -0.5s;
    animation-delay: -0.5s;
}

.sk-fading-circle .sk-circle9:before {
    -webkit-animation-delay: -0.4s;
    animation-delay: -0.4s;
}

.sk-fading-circle .sk-circle10:before {
    -webkit-animation-delay: -0.3s;
    animation-delay: -0.3s;
}

.sk-fading-circle .sk-circle11:before {
    -webkit-animation-delay: -0.2s;
    animation-delay: -0.2s;
}

.sk-fading-circle .sk-circle12:before {
    -webkit-animation-delay: -0.1s;
    animation-delay: -0.1s;
}

@-webkit-keyframes sk-circleFadeDelay {
    0%, 39%, 100% {
        opacity: 0;
    }
    40% {
        opacity: 1;
    }
}

@keyframes sk-circleFadeDelay {
    0%, 39%, 100% {
        opacity: 0;
    }
    40% {
        opacity: 1;
    }
}