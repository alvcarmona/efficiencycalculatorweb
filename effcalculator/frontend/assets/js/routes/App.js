import React, {Component} from 'react';
import Homepage from  './homepage'
import Explanation from './ExplanationComponent'
import detectors from './detectors/index'
import NavBarContainer from './NavbarContainer'
import '../../styles/App.css';
import {Route, Switch, Link} from 'react-router-dom'



class App extends Component {
    constructor(props) {
		super(props);
	}
    render() {
        return (
            <div className="App">

                <div>
                    <NavBarContainer/>
                    <Switch>
                       <Route exact path='/' component={Homepage}/>
                        <Route path='/detectors/' component={detectors} />
                         <Route path='/explanation/' component={Explanation} />
                    </Switch>
                </div>
            </div>
        );
    }
}


export default App;