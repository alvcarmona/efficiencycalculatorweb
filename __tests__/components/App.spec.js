import React from 'react';
import {shallow} from 'enzyme';
import App from '../../effcalculator/frontend/assets/js/routes/App';
import NavBarContainer from '../../effcalculator/frontend/assets/js/routes/NavbarContainer';

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a <div />', () => {
    expect(wrapper.find('.App').length).toEqual(1);
  });
  it('should render the NavBar Component', () => {
    expect(wrapper.containsMatchingElement(<NavBarContainer />)).toEqual(true);
  });
});