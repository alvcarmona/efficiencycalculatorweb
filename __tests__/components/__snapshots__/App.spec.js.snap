// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavbarContainer />,
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/detectors/"
          />
          <Route
            component={[Function]}
            path="/explanation/"
          />
        </Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              path="/detectors/"
            />,
            <Route
              component={[Function]}
              path="/explanation/"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/detectors/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/explanation/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavbarContainer />,
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              path="/detectors/"
            />
            <Route
              component={[Function]}
              path="/explanation/"
            />
          </Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
                path="/detectors/"
              />,
              <Route
                component={[Function]}
                path="/explanation/"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/detectors/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/explanation/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
